### PR TITLE
CV2-4072: Review fields of types `TeamType` and `UserType`

### PR DESCRIPTION
**Pull request by melsawy**
## Description

User PublicTeamType for the following types:

- FeedTeamType
- FeedType
- ProjectGroupType
- ProjectType

Limit fields for UserType  and add a new type `MeType` with current `UserType` fields 

References: CV2-4072

## How has this been tested?

Re-run automated tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

**Review by caiosba — CHANGES_REQUESTED** [review]
Thanks Sawy, I just noticed one more case: `team_author` in `BotUserType` should be `PublicTeamType` too. After this fix, fixes for the tests and changes on Check Web side, I think this is good to go :)

**Comment by caiosba** [comment]
@melsawy also, please be sure it works on the Check Web side, specially with feed features - @danielevalverde can help you with testing this if needed.

File: app/models/user.rb [hunk 1]
```diff
@@ -95,6 +95,10 @@ def self.from_token(token)
     JSON.parse(Base64.decode64(token.gsub('++n', "\n")))
   end
 
+  def me
+    self
```
[the remark this answers is not in the document]
Comment (melsawy) [reply app/models/user.rb:99]: yes @caiosba makes sense and as you know I used this one as a return value for `updateUser` mutation so it'll return a value when current user did  update action.

[the remark this answers is not in the document]
Comment (melsawy) [reply app/models/user.rb]: Done

**Comment by melsawy** [comment]
@caiosba Already I tested the app locally especially features related to changed files.
also I back `medias_count` field to `PublicTeamType` with same way as other count fields like `trash_count`, `unconfirmed_count`, etc as without this field the item does not appear on item page immediately and require a page refresh as the item still assigned to a default project by default.
I think we can review these fields later after deprecate folders from the back-end

**Review by caiosba — APPROVED** [review]

**Pull request merged**